src/effects.ts

```diff
@@ -0,0 +1,203 @@
+        composer.multisampling = 2;
+        break;
+      case AAModes.MSAA_4X:
+        composer.multisampling = 4;
+        break;
+      case AAModes.MSAA_8X:
+        composer.multisampling = 8;
+        break;
+    }
+    console.log(aaPass.enabled, composer.multisampling);
+
+    // either aaPass or copyToScreenPass will handle the final copy to the canvas framebuffer
+    copyToScreenPass.enabled = !aaPass.enabled;
+  }
+
+  const aaMode = store.state.preferences.aaMode;
+  applyAAMode(aaMode);
+
+  // TODO this is a bit of a hack using an empty effect pass to copy buffers so that renderUIPass uses same depth buffer as renderScenePass
+  // There should be a cleaner way to just bind the depth buffer for the renderUIPass correctly.
+  (copyBuffersPass as any).skipRendering = false;
+  copyBuffersPass.needsSwap = true;
+
+  let debugCamera: OrthographicCamera;
+  let debugMeshes: Mesh[];
+  let updateDebugMeshes: () => void;
+  if (qsTruthy("envSettingsDebug") && bloomAndTonemapPass) {
+    const bloom = (bloomAndTonemapPass as any).effects[0] as BloomEffect;
+
+    const debugScene = new Scene();
+    debugCamera = new OrthographicCamera(0, canvas.width, 0, canvas.height, 0.1, 100);
+    debugCamera.layers.enable(Layers.CAMERA_LAYER_FX_MASK);
+    debugCamera.matrixAutoUpdate = true;
+    debugCamera.position.z = 5;
+
+    debugMeshes = [bloom.luminancePass.texture, bloom.texture].map(function (t) {
+      const imageMesh = createImageMesh(t, 1);
+      imageMesh.material.depthTest = false;
+      debugScene.add(imageMesh);
+      return imageMesh;
+    });
+
+    updateDebugMeshes = () => {
+      let y = 10;
+      for (const imageMesh of debugMeshes) {
+        const height = 200;
+        const width = height * (canvas.width / canvas.height);
+        imageMesh.scale.x = width;
+        imageMesh.scale.y = height;
+        imageMesh.position.y = y + height / 2;
+        imageMesh.position.x = 10 + width / 2;
+        imageMesh.matrixNeedsUpdate = true;
+        y += height + 10;
+      }
+    };
+    updateDebugMeshes();
+
+    const debugPass = new RenderPass(debugScene, debugCamera);
+    debugPass.ignoreBackground = true;
+    debugPass.clear = false;
+    debugPass.renderToScreen = true;
+    debugPass.enabled = false;
+
+    composer.addPass(debugPass);
+
+    setTimeout(function () {
+      const envSystem = sceneEl.systems["hubs-systems"].environmentSystem;
+      const gui = envSystem.debugGui.addFolder("Post Effects Debug");
+      {
+        const f = gui.addFolder("Bloom");
+        f.add(bloom.blendMode, "blendFunction", BlendFunction);
+        f.add(bloom.blendMode.opacity, "value", 0, 1).name("Opacity");
+      }
+      gui.add({ aaMode }, "aaMode", AAModes).onChange(applyAAMode);
+      gui.add(debugPass, "enabled").name("debug");
+    }, 0);
+  }
+
+  (sceneEl as any).addEventListener("rendererresize", function ({ detail }: { detail: DOMRectReadOnly }) {
+    composer.setSize(detail.width, detail.height, true);
```

Comment:
Ah yeah good catch. Updated here and in EffectComposer `setSize` call as well. As written its actually a bug since sometimes we want to render at a lower resolution than the canvas is sized to.